app/services/filter_cms.rb

```diff
@@ -0,0 +1,49 @@
+class FilterCms
+  @version = "" # to store version of a url
+  class << self
+
+    def start_filter_wordpress_sites(urls_data)
+      url_html_version_map = Hash.new{ |h,k| h[k] = [] }
+      multi_threads = []
+      urls_data.each do |url|
+        multi_threads <<  Thread.new() {
+          begin
+            html = Nokogiri::HTML.parse(RestClient.get url)
```

Comment:
`RestClient.get` supports use of proxies, research on how to use proxies in RestClient ruby to get the required URL